modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk/query_processor.clj

```diff
@@ -604,9 +605,10 @@
 (doseq [filter-type [:between := :!= :> :>= :< :<=]]
   (defmethod sql.qp/->honeysql [:bigquery-cloud-sdk filter-type]
     [driver clause]
-    ((get-method sql.qp/->honeysql [:sql filter-type])
-     driver
-     (reconcile-temporal-types clause))))
+    (reconcile-temporal-types
+     ((get-method sql.qp/->honeysql [:sql filter-type])
+      driver
+      (reconcile-temporal-types clause)))))
```

Comment:
Interesting, reconciling twice.

Reply:
Doesn't really hurt anything

Reply:
In which cases is it useful? (I'm asking because I suppose you have cases in mind I haven't though about.)

Reply:
It wouldn't work if we didn't reconcile after compilation. Not sure if the initial reconciliation before compilation is still needed or not. I didn't check. You can try taking it out and seeing if it works or not

Reply:
Intuitively, I would think it's an abuse of reconciliation if we have to do it twice. I might be wrong though.